### ISSUE TITLE
v0.5.0 - generation of zsh completions fails

### ISSUE THREAD

**Issue by bket** [rustic-rs/rustic#533]
As of v0.5.0 rustic (release build) fails generating zsh completions. Interestingly is that generation of bash and fish completions succeeds. 

```
$ RUST_BACKTRACE=full ./rustic completions zsh 
using no config file (./rustic.toml doesn't exist)
thread 'main' panicked at 'Command::get_arg_conflicts_with: The passed arg conflicts with an arg unknown to the cmd', /home/bket/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.2.23/src/builder/command.rs:3604:21
stack backtrace:
   0:      0x3b21242147a - <unknown>
   1:      0x3b21243f99a - <unknown>
   2:      0x3b2123f96d5 - <unknown>
   3:      0x3b212421250 - <unknown>
   4:      0x3b2123fb0a5 - <unknown>
   5:      0x3b2123fad4e - <unknown>
   6:      0x3b2123fbb06 - <unknown>
   7:      0x3b212422266 - <unknown>
   8:      0x3b2124215c0 - <unknown>
   9:      0x3b2123fb675 - <unknown>
  10:      0x3b212442764 - <unknown>
  11:      0x3b21228275a - <unknown>
  12:      0x3b211f1cdd8 - <unknown>
  13:      0x3b211f1b155 - <unknown>
  14:      0x3b211f19b30 - <unknown>
  15:      0x3b211f18d69 - <unknown>
  16:      0x3b211d19aa9 - <unknown>
  17:      0x3b211d4d990 - <unknown>
  18:      0x3b211d2b336 - <unknown>
  19:      0x3b211cba0e1 - <unknown>
  20:      0x3b2123fd9c8 - <unknown>
  21:      0x3b211c17c3d - <unknown>
  22:      0x3b211b75702 - <unknown>
```

To get some usefull information on this panic I tested an unoptimized + debuginfo build, which seems to follow a different codepath resulting in a whole new issue. Using this build also generation of bash and fish completions fail with the panic below.

```
$ RUST_BACKTRACE=full ./rustic completions zsh
using no config file (./rustic.toml doesn't exist)
thread 'main' panicked at 'Command backup: Short option names must be unique for each argument, but '-g' is in use by both 'group-by' and 'glob'', /home/bket/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.2.23/src/builder/debug_asserts.rs:124:17
stack backtrace:
   0:      0x8f3d1774d2a - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::ha21eed264b1f83e5
   1:      0x8f3d1793cba - core::fmt::write::h15be51aa36c4b2ff
   2:      0x8f3d174cf85 - std::io::Write::write_fmt::h98c87d58439337ed
   3:      0x8f3d1774b00 - std::sys_common::backtrace::print::h796e6d8e17faeaf8
   4:      0x8f3d174e955 - std::panicking::default_hook::{{closure}}::h5770a9fa2db9d7ff
   5:      0x8f3d174e5fe - std::panicking::default_hook::h4bb6b6dda26b07c0
   6:      0x8f3d174f3b6 - std::panicking::rust_panic_with_hook::hedeafd98477558ba
   7:      0x8f3d1775b5c - std::panicking::begin_panic_handler::{{closure}}::hf77787b86bdcb448
   8:      0x8f3d1774e70 - std::sys_common::backtrace::__rust_end_short_backtrace::h774ea7e9d01034e8
   9:      0x8f3d174ef25 - rust_begin_unwind
  10:      0x8f3d1796a94 - core::panicking::panic_fmt::h47555f72596e3c42
  11:      0x8f3d12d579e - clap::builder::debug_asserts::assert_app::h201c8611eaf4353a
                               at /home/bket/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.2.23/src/builder/debug_asserts.rs:124:17
  12:      0x8f3d123fe02 - clap::builder::command::App::_build_self::hd7ad8850155ab70a
                               at /home/bket/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.2.23/src/builder/command.rs:4340:13
  13:      0x8f3d123faf6 - clap::builder::command::App::_build_recursive::hb686837789f7f91a
                               at /home/bket/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.2.23/src/builder/command.rs:4257:9
  14:      0x8f3d123fb4d - clap::builder::command::App::_build_recursive::hb686837789f7f91a
                               at /home/bket/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.2.23/src/builder/command.rs:4259:13
  15:      0x8f3d123fac5 - clap::builder::command::App::build::hbd80420e2fc0fae2
                               at /home/bket/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-3.2.23/src/builder/command.rs:4252:9
  16:      0x8f3d02fc389 - clap_complete::generator::_generate::h1061eb23aac1a70b
                               at /home/bket/.cargo/registry/src/github.com-1ecc6299db9ec823/clap_complete-3.2.5/src/generator/mod.rs:239:5
  17:      0x8f3d02fc242 - clap_complete::generator::generate::h9f686c9341ccd5cc
                               at /home/bket/.cargo/registry/src/github.com-1ecc6299db9ec823/clap_complete-3.2.5/src/generator/mod.rs:231:5
  18:      0x8f3d040f844 - rustic::commands::completions::generate_completion::ha97cb4723bb29dd7
                               at /home/code/rustic/src/commands/completions.rs:27:5
  19:      0x8f3d02faa40 - rustic::commands::completions::execute::hada0ea84e4d648d2
                               at /home/code/rustic/src/commands/completions.rs:21:25
  20:      0x8f3d0270729 - rustic::commands::execute::hc23536c2d3b40338
                               at /home/code/rustic/src/commands/mod.rs:218:9
  21:      0x8f3d02e2788 - rustic::main::h74fe063cf47b2943
                               at /home/code/rustic/src/main.rs:61:5
  22:      0x8f3d02866be - core::ops::function::FnOnce::call_once::h7297d0ad100e287e
                               at /usr/obj/ports/rust-1.68.0/rustc-1.68.0-src/library/core/src/ops/function.rs:250:5
  23:      0x8f3d0366e41 - std::sys_common::backtrace::__rust_begin_short_backtrace::hbb6667e836675117
                               at /usr/obj/ports/rust-1.68.0/rustc-1.68.0-src/library/std/src/sys_common/backtrace.rs:121:18
  24:      0x8f3d0188114 - std::rt::lang_start::{{closure}}::h24eb72bf8cd84977
                               at /usr/obj/ports/rust-1.68.0/rustc-1.68.0-src/library/std/src/rt.rs:166:18
  25:      0x8f3d1751278 - std::rt::lang_start_internal::h125fa9c97190146f
  26:      0x8f3d01880e8 - std::rt::lang_start::h7d8fab1c963af15a
                               at /usr/obj/ports/rust-1.68.0/rustc-1.68.0-src/library/std/src/rt.rs:165:17
  27:      0x8f3d02e27b1 - main
  28:      0x8f3d0015d42 - <unknown>
```

**Comment by aawsome** [comment]
Thanks a lot for reporting! Funny that both `-g` clap options exist for more than 4 month now...
Maybe a new version of the completions generator got more pedantic here.

Anyway, I'll add the creation of completions to the CI tests and of course fix this!

**Comment by aawsome** [comment]
When adding the completions to the automatic tests, I found a few other issues. They should be all fixed within #536.
